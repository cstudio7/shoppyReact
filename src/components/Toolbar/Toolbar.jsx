import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import AdminNav from '../AdminNav';
import DrawerToggleButton from '../SideDrawer/DrawerToggleButton';
import './Toolbar.scss';

const toolbar = props => (
  <header className="toolbar">
    <nav className="toolbar__navigation">
      <div className="toolbar__toggle-button">
        <DrawerToggleButton click={props.drawerClickHandler} />
      </div>
      <div className="toolbar__logo">
        <Link to="/login">Shoppy</Link>
      </div>
      <div className="spacer" />
      <div className="toolbar_navigation-items">
        <AdminNav />
      </div>
    </nav>
  </header>
);

toolbar.propTypes = {
  drawerClickHandler: PropTypes.func.isRequired
};

export default toolbar;
