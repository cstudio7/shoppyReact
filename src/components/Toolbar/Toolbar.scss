.toolbar {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  background: #ff6300;
  height: 56px;
}

.toolbar__navigation {
  display: flex;
  height: 100%;
  align-items: center;
  padding: 0 1rem;
}

.toolbar__logo {
  margin-left: 1rem;
}

.toolbar__logo a {
  color: white;
  text-decoration: none;
  font-size: 1.5rem;
}

.spacer {
  flex: 1;
}

.toolbar_navigation-items ul {
  list-style: none;
  margin: 0;
  padding: 0;
  display: flex;
}

.toolbar_navigation-items li {
  padding: 0 0.5rem;
}

.toolbar_navigation-items a {
  color: white;
  text-decoration: none;
}

.toolbar_navigation-items a,
.toolbar_navigation-items a {
  display: inline-block;
  text-decoration: none;
  padding: 13px 23px;
  font-size: 18px;
  -webkit-transition: 0.1s all;
  transition: 0.5s all;
  &:hover {
    color: #ff6300;
    background: #fff;
  }
  &:active {
    color: #ff6300;
    background: #fff;
  }
}

@media (max-width: 768px) {
  .toolbar_navigation-items {
    display: none;
  }
}

@media (min-width: 769px) {
  .toolbar__toggle-button {
    display: none;
  }

  .toolbar__logo {
    margin-left: 0;
  }
}
