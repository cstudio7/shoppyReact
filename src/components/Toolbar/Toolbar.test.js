import React from 'react';
import { shallow } from 'enzyme';
import Toolbar from './Toolbar';

describe('Test the SideDrawer Component', () => {
  it('should match snapshot', () => {
    const drawerClickHandler = jest.fn;
    const userRole = 'attendant';
    const wrapper = shallow(
      <Toolbar drawerClickHandler={drawerClickHandler} userRole={userRole} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
