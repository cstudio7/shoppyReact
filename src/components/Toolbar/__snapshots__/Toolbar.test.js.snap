// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the SideDrawer Component should match snapshot 1`] = `
<header
  className="toolbar"
>
  <nav
    className="toolbar__navigation"
  >
    <div
      className="toolbar__toggle-button"
    >
      <drawerToggleButton
        click={[Function]}
      />
    </div>
    <div
      className="toolbar__logo"
    >
      <Link
        replace={false}
        to="/login"
      >
        Shoppy
      </Link>
    </div>
    <div
      className="spacer"
    />
    <div
      className="toolbar_navigation-items"
    >
      <AdminNav />
    </div>
  </nav>
</header>
`;
